### DISCUSSION TITLE
How to resume a disconnected/reconnected subscription with new parameters?

### DISCUSSION THREAD

**Discussion by mgreystone** [trpc/trpc#4127] [Q&A]
TL;DR: How do you _resume_ an interrupted subscription?

---

I'm looking to perform an operation on a subscription when a websocket reconnects.

I have clients subscribing to a stream of events. I am sending the id of the last event received; my server will only emit events created after the given event id.

```ts
let lastEventId = getFromLocalState()

client.join.subscribe({ lastEventId }, {
  onData(event) {
    lastEventId = event.id
  },
})
```

However, whenever the websocket reconnects, the same `lastEventId` value that was used when the subscription was initially created is re-sent to the server. My server then re-sends events that we have already received. This occurs both when my server restarts & sends `broadcastReconnectNotification` as well as when the websocket is interrupted client-side.

My thought was that subscriptions needed an `onReconnecting` handler so that re-subscriptions like this could be handled. I opened a feature request at https://github.com/trpc/trpc/issues/4122 & started to work on an implementation. But as i got into the weeds, it became clear that there is a strict separation between a subscription & its underlying `wsLink`. I don't see a clear way for `wsLink` to send an event/notification up to any affected subscriptions.

I see a similar issue at https://github.com/trpc/trpc/issues/1574 as well as a PR at https://github.com/trpc/trpc/pull/1584 by @jlalmes . This proposal would accept a function as subscription input that gets re-invoked any time a websocket opens. It looks like it was abandoned, however. Any chance that it could be resurrected? 

Thanks for everything.

**Comment by Elvincth** [comment]
Same issue here, any update on this?
